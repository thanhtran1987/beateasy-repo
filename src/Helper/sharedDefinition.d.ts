interface IEventType {
  EventTypeDesc: string;
  EventTypeID: number;
  MasterEventTypeID: number;
  Slug: string;
}
interface IVenue {
  Venue: string;
  StateCode: string;
  Slug: string;
}

interface IEvent {
  EventID: number;
  MasterEventID: number;
  EventName: string;
  EventTypeDesc: string;
  MasterEventName: string;
  AdvertisedStartTime: string;
  RaceNumber: number;
  EventType: IEventType;
  Venue: IVenue;
  IsMultiAllowed: boolean;
  Slug: string;
  DateSlug: string;
  RacingStreamAllowed: boolean;
  RacingReplayStreamAllowed: boolean;
  Ticks?: string;
}
