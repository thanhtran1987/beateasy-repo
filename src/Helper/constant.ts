// template action for basic async request
const genericAction = (data: string) => {
  return {
    REQUEST: data + '_REQUEST',
    SUCCESS: data + '_SUCCESS',
    ERROR: data + '_ERROR',
  };
};

// basic async action below
export const next5RaceActionType = genericAction('NEXT_5_RACE');

// advance async action below

// sychronious action below
export const pageActionType = {
  SETFILTER: 'SETFILTER',
};
