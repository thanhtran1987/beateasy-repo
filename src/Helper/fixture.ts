export const raceItem = {
  EventID: 30431079,
  MasterEventID: 1049998,
  EventName: 'Lucky Creed',
  EventTypeDesc: 'Trots',
  MasterEventName: 'Races',
  AdvertisedStartTime: '2018-10-08T12:29:30Z',
  RaceNumber: 3,
  EventType: {
    EventTypeID: 2,
    EventTypeDesc: 'Trots',
    MasterEventTypeID: 1,
    Slug: 'harness-racing',
  },
  Venue: { Venue: 'Albion Park', StateCode: 'QLD', Slug: 'albion-park' },
  IsMultiAllowed: true,
  Slug: 'lucky-creed-qualifying-pace',
  DateSlug: '20180910',
  RacingStreamAllowed: false,
  RacingReplayStreamAllowed: false,
};
