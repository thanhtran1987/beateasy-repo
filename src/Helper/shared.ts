export const getCountDown = (data: any): string => {
  if (data < 1) {
    return 'Expired';
  }

  const days = Math.floor(data / 86400);
  data -= days * 86400;
  const hours = Math.floor(data / 3600) % 24;
  data -= hours * 3600;
  const minutes = Math.floor(data / 60) % 60;
  data -= minutes * 60;
  const seconds = Math.floor(data) % 60;

  if (days) {
    return days + 'd ' + hours + 'h';
  } else if (hours) {
    return hours + 'h ' + minutes + 'm';
  } else if (minutes) {
    return minutes + 'm ' + seconds + 's';
  } else {
    return seconds + 's';
  }
};
