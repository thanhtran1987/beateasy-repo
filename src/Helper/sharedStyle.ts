import styled from 'styled-components';

export const COLOR_PRIMARY = '65309c';
export const COLOR_SECONDARY = '253b80';
export const COLOR_ALERT = 'ff4b3d';
export const COLOR_GREY = 'ababab';

export const FlexWrapper = 'ababab';

export const LoaderWrapper = styled.div`
  text-align: center;
  font-size: 12px;
`;

export const MenuWrapper = styled.div`
  border-left: 1px solid #${COLOR_GREY};
  margin-left: 30px;
  text-align: center;
  padding-left: 19px;
  flex-grow: 1;
  ul {
    display: flex;
    height: 80px;
    align-items: center;
    padding: 0;
    margin: 0;
    list-style: none;
    li {
      flex: 0 1 108px;
      span {
        border: solid 1px #${COLOR_GREY}
        border-radius: 50%;
        display: inline-block;
        svg {
          display: block
          margin: 0;
        }
      }
      h1 {
        font-size: 14px;
        margin: 0;
      }
    }
  }
`;
