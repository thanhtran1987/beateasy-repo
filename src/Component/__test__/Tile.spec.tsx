import * as React from 'react';
import { shallow } from 'enzyme';

import TestComponent from '../Tile';

import { raceItem } from '../../Helper/fixture';

describe('Tile', () => {
  it('should render correctly in default state', () => {
    const component = shallow(<TestComponent {...raceItem} />);
    expect(component).toMatchSnapshot();
  });
});
