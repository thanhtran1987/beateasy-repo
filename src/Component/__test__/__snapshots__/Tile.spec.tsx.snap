// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile should render correctly in default state 1`] = `
<styled.div>
  <styled.div>
    <Component />
  </styled.div>
  <styled.div>
    <div>
      <styled.div>
        Lucky Creed
      </styled.div>
      <styled.div>
        Race 
        3
      </styled.div>
    </div>
    <styled.div>
      Expired
    </styled.div>
  </styled.div>
</styled.div>
`;
